### PR TITLE
Hw docker

### PR DESCRIPTION
**Pull request by GradientAspen**

### CODE REVIEW

**Comment by GradientAspen** [comment]
I have a problem connecting to the debugger. During startup, the error "Unable to open debugger port (localhost:8000): java.io.IOException "handshake failed - connection prematurally closed""

File: src/main/java/com/project/springbootproject/repository/cartitem/CartItemRepository.java [hunk 1]
```diff
@@ -2,9 +2,11 @@
 
 import com.project.springbootproject.model.CartItem;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
 import org.springframework.data.jpa.repository.Query;
 
 public interface CartItemRepository extends JpaRepository<CartItem, Long> {
+    @Modifying
     @Query("UPDATE CartItem ci SET ci.isDeleted = true WHERE ci.shoppingCart.id = :id")
     void deleteCartItemsFromDb(Long id);
```
Comment (okuzan) [inline src/main/java/com/project/springbootproject/repository/cartitem/CartItemRepository.java:11]: if you used annotations correctly, it should be done automatically

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,4 +1,5 @@
-spring.datasource.url=jdbc:mysql://localhost:3306/bookDb?serverTimeZone=UTC
+#spring.datasource.url=jdbc:mysql://localhost:3306/bookDb?serverTimeZone=UTC
```
Comment (okuzan) [inline src/main/resources/application.properties]: always remove commented out code

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+version: '3.8'
+
+services:
+  myapp:
+    build: .
+    ports:
+      - "8080:8080"
+      - "8000:8000"
+    environment:
```
Comment (okuzan) [inline docker-compose.yml]: incomplete configuration

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/com/project/springbootproject/repository/cartitem/CartItemRepository.java (hunk 1 above)
Comment (okuzan) [reply src/main/java/com/project/springbootproject/repository/cartitem/CartItemRepository.java:11]: not fixed

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/com/project/springbootproject/repository/cartitem/CartItemRepository.java (hunk 1 above)
Comment (GradientAspen) [reply src/main/java/com/project/springbootproject/repository/cartitem/CartItemRepository.java:11]: An error occurs when using the method without annotation 2024-03-05T12:41:35.951+02:00 ERROR 22868 --- [nio-8080-exec-5] o.a.c.c.C.[.[.[.[dispatcherServlet]      : Servlet.service() for servlet [dispatcherServlet] in context with path [/api] threw exception [Request processing failed: org.springframework.dao.InvalidDataAccessApiUsageException: Expecting a SELECT query : `UPDATE CartItem ci SET ci.isDeleted = true WHERE ci.shoppingCart.id = :id`] with root cause

org.hibernate.query.IllegalSelectQueryException: Expecting a SELECT Query [org.hibernate.query.sqm.tree.select.SqmSelectStatement], but found org.hibernate.query.sqm.tree.update.SqmUpdateStatement [UPDATE CartItem ci SET ci.isDeleted = true WHERE ci.shoppingCart.id = :id]
	at org.hibernate.query.sqm.internal.SqmUtil.verifyIsSelectStatement(SqmUtil.java:93) ~[hibernate-core-6.3.1.Final.jar:6.3.1.Final]
	at org.hibernate.query.sqm.internal.QuerySqmImpl.verifySelect(QuerySqmImpl.java:489) ~[hibernate-core-6.3.1.Final.jar:6.3.1.Final]
	at org.hibernate.query.sqm.internal.QuerySqmImpl.doList(QuerySqmImpl.java:498) ~[hibernate-core-6.3.1.Final.jar:6.3.1.Final]
	at org.hibernate.query.spi.AbstractSelectionQuery.list(AbstractSelectionQuery.java:427) ~[hibernate-core-6.3.1.Final.jar:6.3.1.Final]
	at org.hibernate.query.spi.AbstractSelectionQuery.getSingleResult(AbstractSelectionQuery.java:559) ~[hibernate-core-6.3.1.Final.jar:6.3.1.Final]
	at org.springframework.data.jpa.repository.query.JpaQueryExecution$SingleEntityExecution.doExecute(JpaQueryExecution.java:223) ~[spring-data-jpa-3.2.0.jar:3.2.0]

**Review by okuzan — APPROVED** [review]

**Pull request merged**